### ISSUE TITLE
Continuous action spaces for RL

### ISSUE THREAD

**Issue by peastman** [deepchem/deepchem#983]
Currently all the reinforcement learning classes only support environments with discrete action spaces.  It would be useful to support continuous action spaces as well.

**Comment by rbharath** [comment]
+1. This would make the RL code a good bit more useful.

**Comment by peastman** [comment]
To do this, we need to be able to provide three calculations:

1. Given the output of the policy, select an action by sampling from some distribution.

2. Given the action that was selected, calculate what the probability was of selecting that action.  This needs to be implemented in Tensorflow so that it can backpropagate through it.  It needs to be able to compute derivatives of the probability with respect to the model parameters.

3. A3C and PPO each include an entropy term in the loss, so we need to compute the entropy of the action distribution given the policy output.

I could try to implement all of this in a generic way.  That would probably take a lot of new infrastructure.  Alternatively I could just assume a gaussian distribution, which is what's used in all the examples I've found.  The policy would output a set of means and a set of variances, and we would select the action based on them.

**Comment by rbharath** [comment]
I like the idea of using Gaussians. It's a lot simpler and probably good enough for most real world applications.

**Comment by rbharath** [comment]
This may potentially be relevant. It's a collection of robotic control algorithms open sourced by deep-mind and might be useful for ideas:

https://github.com/deepmind/dm_control
